### PR TITLE
Use `COOKIES_FILE` to fetch page titles

### PR DESCRIPTION
**Pull request by benmuth**
# Summary
This PR lets the title extractor use the `COOKIES_FILE`, if available. This helps avoid extracting the titles of captcha or login pages.

# Related issues
#761 

# Changes these areas

- [ ] Bugfixes
- [x] Feature behavior
- [ ] Command line interface
- [x] Configuration options
- [ ] Internal architecture
- [ ] Snapshot data layout on disk

### CODE REVIEW

File: archivebox/util.py [hunk 1]
```diff
@@ -164,13 +165,26 @@ def parse_date(date: Any) -> Optional[datetime]:
...
+    from .config import (
+         TIMEOUT,
+         CHECK_SSL_VALIDITY,
+         WGET_USER_AGENT,
+         COOKIES_FILE,
+    )
     timeout = timeout or TIMEOUT
+
+    cookie_jar = http.cookiejar.MozillaCookieJar()
+    if COOKIES_FILE is not None:
```
Comment (pirate) [inline archivebox/util.py:177]:
```suggestion
    if COOKIES_FILE and Path(COOKIES_FILE).is_file():
```

**Comment by pirate** [comment]
Down to merge this if you've tested it! Just change it to `Ready for Review` when you're ready :)

**Comment by benmuth** [comment]
I've been testing it and I'm actually having trouble verifying that this works. I thought it was working when I opened this PR, but it seems I was wrong. For my test case, the title isn't correct if the title extractor is the only enabled extractor. It is correct when `wget` is enabled, so the cookie file is being read correctly by ArchiveBox, but my fix here doesn't seem to be working. Not sure where it's failing yet.

**Comment by benmuth** [comment]
After throwing some stuff at the wall, I got to a solution that works for me. For some reason, the cookie file only seems to work with `requests` when I set each cookie one by one, like 

```
        for cookie in cookie_jar:
            session.cookies.set(cookie.name, cookie.value, domain=cookie.domain, path=cookie.path)
```
It didn't work for me when I tried `session.cookies = cookie_jar` or `session.cookies.update(cookie_jar)`, and from the docs I can't tell why. If anyone could tell me what I'm missing, I'd appreciate it.

**Pull request merged**